Chore: #5 프로젝트 초기 세팅

MuchanKim opened the pull request:
<!--
🙏 PR 제목 컨벤션 (Gitmoji + 타입 + 이슈 번호 + 작업 요약)
예시: ✨ Feat: #167 예약 취소 구현
※ PR 생성 시 Assignees 및 Labels 설정도 잊지 마세요!
-->

## ✨ What’s this PR?
### 📌 관련 이슈 (Related Issue)
<!-- 해당 PR이 어떤 이슈를 해결하는지 연결해주세요 -->
- Closes #5 

---

### 🧶 주요 변경 내용 (Summary)
<!-- 이번 PR에서 작업한 핵심 변경 사항을 작성해주세요 -->

```
📁 NoMyeolmang Project
├── 📁 NoMyeolmang-iOS (iOS 그룹)
│ ├── Sources/
│ │ ├── App/
│ │ └── Presentations/
│ └── Resources/
│
├── 📁 NoMyeolmang-watch (watchOS 그룹)
│ ├── Sources/
│ │ ├── App/
│ │ └── Presentations/
│ └── Resources/
│
└── 📁 Shared (공통 그룹)
├── ML/
├── Models/
└── Services/
```

- iOS, watchOS 타겟 설정 완료했습니다.
- 각 타겟 별 초기 폴더 구조 생성 및 샘플 파일 추가했습니다.
- 구조에 관하여... 2 target을 가지는 프로젝트가 처음이라 구조가 바로 떠오르지 않았습니다. 일단, share를 중앙 처리 롤을 맡게 하고, watch, ios에서 요청하는 방식을 떠올렸습니다.
- swiiftlint 적용
- router 클래스 구현(dodin)

---

### 📸 스크린샷 (Optional)
<!-- UI 작업의 경우, 구현한 화면을 첨부해주세요 -->
<!-- 이미지 크기 조절 예시: <img width="300" alt="설명" src="링크"> -->

---

### 🧪 테스트 / 검증 내역
<!-- 동작 확인 여부나 시나리오 테스트 내용을 간단히 써주세요 -->

- [x] iOS, watchOS 타겟 빌드 및 실행 확인
- [x] Shared 폴더 내 모델/서비스 import 정상 동작

---

### 💬 기타 공유 사항
<!-- 리뷰어가 참고하면 좋을 정보, 고민했던 지점 등을 적어주세요 -->
[팀노션 - 폴더구조](https://www.notion.so/posacademy/22b2b843d5af809ca0e4e749babe3281?source=copy_link)
[팀노션 - swiftlint 적용법](https://www.notion.so/posacademy/Swiftlint-22c2b843d5af80fc8299f441ab82e421?source=copy_link)

위 글 참고해주세요.

---

### 🙇🏻‍♀️ 리뷰 가이드 (선택)
<!-- 리뷰어가 중점적으로 보면 좋을 포인트가 있다면 알려주세요 -->

피드백 받습니다. 
- 폴더 구조 및 타겟 멤버십 설정 방식
- Shared 폴더 활용 방식

MuchanKim committed:
del: #5 불필요한 파일 제거

- swiftlint 파일은 프로젝트 파일 추가 시 함께 올리겠습니다.
- Conduct, Contributing 파일은 필요 없다고 판단되어 제거합니다.

MuchanKim committed:
del: #5 LICENSE 파일 제거

- LICENSE 파일을 제거합니다.

ohdodin committed:
chore: #5 이슈&PR 템플릿 수정

ohdodin committed:
chore: #5 YAML 문법 맞춰서 수정

- 여백으로 인한 오류 발생
- general 파일 수정 덜됨 (죄송)

MuchanKim committed:
chore: #5 프로젝트 초기 세팅

- share에 서비스/비지니스 로직을 포함하는 구조로 변경했습니다. (노션 확인)

- target 설정 확인 완료

- share 폴더 target membership 설정 완료

MuchanKim committed:
Merge branch 'chore/#5/initial-setting' of https://github.com/DeveloperAcademy-POSTECH/2025-C4-M9-NoMyeolmang into chore/#5/initial-setting

MuchanKim committed:
fix: #5 gitignore 긴급 수정

- xcodeproj가 무시되는 현상 발생에 따른 수정

MuchanKim committed:
add: #5 xcodeproj 파일 추가

- #7에 따른 문제 해결.

Commit:
Feat: Navagation Route 패턴 적용

- RootView 생성 및 @main에 첫화면으로 보이도록 설정
- AppRoute, AppRouter 정의
- HomeView, HomeViewModel에 예시로 navigation 코드 적용

ohdodin reviewed (approved):
Good.

MuchanKim commented on NoMyeolmang/NoMyeolmang-iOS/Sources/App/AppRouter.swift:
```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+final class AppRouter: ObservableObject {
+    @Published var path: [AppRoute] = []
```
navigationPath() 대신 스택을 쓰신 이유가 있을까요?

ohdodin replied:
navigationPath로 수정은 했는데 이 방법을 써야 더 좋은 점이 있나요?

MuchanKim replied:
오호! 수정하라는 의미는 아니었어요! 그냥 궁금해서!

MuchanKim commented on NoMyeolmang/NoMyeolmang-iOS/Sources/App/RootView.swift:
```diff
@@ -0,0 +1,29 @@
+import SwiftUI
+
+struct RootView: View {
+    @StateObject var router = AppRouter()
```
AppRouter는 View마다 주입되어야하는 클래스입니다. StateObject로  선언한 이유가 있을까요? EnviromentObject로 선언하여 전역 인스턴스로 주입하는 방식도 있습니다.

ohdodin replied:
수정했습니다.

MuchanKim commented on NoMyeolmang/NoMyeolmang-iOS/Sources/App/RootView.swift:
```diff
@@ -0,0 +1,29 @@
+
+    var body: some View {
+        NavigationStack(path: $router.path) {
+            HomeView(viewModel: HomeViewModel(router: router))
+                .navigationDestination(for: AppRoute.self) { route in
+                    switch route {
+                    case .home:
+                        HomeView(viewModel: HomeViewModel(router: router))
```
뷰모델과 네비게이션 로직은 분리되어야합니다.

ohdodin replied:
분리시켰습니다

MuchanKim committed:
chore: #5 잘못들어간 코드 제거

- lint 실험하다가 잘못된 코드가 들어갔네요.

MuchanKim committed:
chore: #5 swiftlint 추가

- 적용 확인했습니다.

Commit:
Refactor: navigation 코드 수정

- navigation과 viewModel 의존성 분리
- AppRotuer environmentObject로 주입
- navigationPath 사용

Commit:
Merge branch 'chore/#5/initial-setting' of https://github.com/DeveloperAcademy-POSTECH/2025-C4-M9-NoMyeolmang into chore/#5/initial-setting

MuchanKim committed:
chore: #5 swiftlint.yml 수정

- swift style guide 규칙을 준수하는 구문 추가

Commit:
refact: navigation 코드 수정

- test 후 작동하도록 view, viewmodel 코드 수정
- appicon 충돌 발생하여 -iOS, -watch 추가

Commit:
fix: AppIcon 이름 수정 취소

- 다른 파일 섞여서 오류가 나는 거였습니다.. (죄송)
- 다시 AppIcon으로 이름 수정
